service/history/ndc/workflow_state_replicator.go

```diff
@@ -694,6 +694,9 @@ func (r *WorkflowStateReplicatorImpl) getNewRunMutableState(
 
 	if isStateBased {
 		newRunMutableState.InitTransitionHistory()
+		for _, event := range newRunHistory {
+			newRunMutableState.AddReapplyCandidateEvent(event)
+		}
```

Comment:
This new run (continue as new) is created by applying events so the history builder in newRunMutableState does contain those events I think.